rules/group.go

```diff
@@ -621,7 +621,16 @@ func (g *Group) cleanupStaleSeries(ctx context.Context, ts time.Time) {
 		}
 	}
 	if err := app.Commit(); err != nil {
-		level.Warn(g.logger).Log("msg", "Stale sample appending for previous configuration failed", "err", err)
+		switch {
```

Comment:
We experienced this issue in Mimir, and due to how it works (a distributed system) we detect these errors in the `Commit()` rather than `Append()`. However, for the Prometheus case, all such errors should be actually detected by the `Append()` so I'm not sure this change is correct in the Prometheus' context.

_I will follow up with you offline._

Reply:
I also think in the prometheus context this does not look correct

Reply:
@pracucci any updates from that follow up?

@bboreham and I have looked at this during the prometheus bug scrub and also agree that this error should not be detected in Commit(). Also, a few lines above we have the detection the append section where its already being detected and logged of explicitely.

Reply:
> @pracucci any updates from that follow up?

I suggested @jhesketh to fix it in Mimir instead.